pkg/artifactory/resource_artifactory_pull_replication.go

```diff
@@ -27,9 +27,9 @@ func resourceArtifactoryPullReplication() *schema.Resource {
 	}
 }
 
-func unpackPullReplication(s *schema.ResourceData) *utils.ReplicationBody {
+func unpackPullReplication(s *schema.ResourceData) *utils.UpdateReplicationBody {
 	d := &ResourceData{s}
-	replicationConfig := new(utils.ReplicationBody)
+	replicationConfig := new(utils.UpdateReplicationBody)
```

Comment:
So, not your fault, but this is mixing types from the JFROG client go and using some locals too. I would prefer if you literally C&P the `utils.*` into the file and ditch the dependency. I'll let @alexhung chime in

Reply:
Agree, @chb0github. As first step of removing dependency on `jfrog-client-go`, we should do what you describe.

Reply:
Ok, sounds good.  I'll do as chb0github suggests and copy out the types into this repo and not add more dependencies on `jfrog-client-go`